### ISSUE TITLE
Projectile Turrets

### ISSUE THREAD

**Issue by NovaZHart** [NovaZHart/ShooterPrototype#1]
AI for ship-mounted turrets using ProjectileTurrent class. 

* opportunistic firing: turret points outward from ship, waiting for a nearby target. Aim at target of opportunity and fire.
* targeted firing: turrets all point towards ship's current target

~~To avoid writing this code twice, #3 should be done first.~~

**Note**: projectile code will have to be rewritten after #18, but weapon logic won't suffer as much change.

**Comment by NovaZHart** [comment]
Closing issue, will open another issue for opportunistic firing.
